packages/create-bison-app/template/src/env.mjs

```diff
@@ -0,0 +1,96 @@
+import { z } from 'zod';
+
+const nodeEnv = process.env.NODE_ENV || 'development';
+
+/**
+ * Specify your server-side environment variables schema here. This way you can ensure the app isn't
+ * built with invalid env vars.
+ */
+const server = z.object({
+  APP_ENV: z.enum(['development', 'test', 'production']),
+  APP_URL: z.string(),
+  DATABASE_URL: z.string().url(),
+  DEBUG: z.boolean().optional(),
+  FC_GIT_COMMIT_SHA: z.string().optional(),
+  IS_CI: z.boolean().optional(),
+  NEXTAUTH_SECRET:
+    process.env.NODE_ENV === 'production' ? z.string().min(1) : z.string().min(1).optional(),
+  NEXTAUTH_URL: z.string().min(1),
+  NODE_ENV: z.enum(['development', 'test', 'production']),
+  PORT: z.number().int().positive(),
+  RENDER_EXTERNAL_URL: z.string().optional(),
+  SHOULD_MIGRATE: z.boolean(),
+});
+
+/**
+ * Specify your client-side environment variables schema here. This way you can ensure the app isn't
+ * built with invalid env vars. To expose them to the client, prefix them with `NEXT_PUBLIC_`.
+ */
+const client = z.object({
+  NEXT_PUBLIC_APP_ENV: z.enum(['development', 'test', 'production']),
+});
+
+/**
+ * You can't destruct `process.env` as a regular object in the Next.js edge runtimes (e.g.
+ * middlewares) or client-side so we need to destruct manually.
+ *
+ * @type {Record<keyof z.infer<typeof server> | keyof z.infer<typeof client>, string | undefined>}
+ */
+const processEnv = {
+  APP_ENV: process.env.APP_ENV || nodeEnv,
+  APP_URL: process.env.APP_URL || 'http://localhost:3000',
+  DATABASE_URL: process.env.DATABASE_URL,
+  DEBUG: process.env.DEBUG || false,
+  FC_GIT_COMMIT_SHA: process.env.FC_GIT_COMMIT_SHA,
+  IS_CI: process.env.CI || false,
+  NEXT_PUBLIC_APP_ENV: process.env.NEXT_PUBLIC_APP_ENV || nodeEnv,
+  NEXTAUTH_SECRET: process.env.NEXTAUTH_SECRET,
+  NEXTAUTH_URL: process.env.NEXTAUTH_URL,
+  NODE_ENV: nodeEnv,
+  PORT: process.env.PORT || 3000,
+  RENDER_EXTERNAL_URL: process.env.RENDER_EXTERNAL_URL,
+  SHOULD_MIGRATE: process.env.SHOULD_MIGRATE || false,
+};
+
+// Don't touch the part below
+// --------------------------
+
+const merged = server.merge(client);
+
+/** @typedef {z.input<typeof merged>} MergedInput */
+/** @typedef {z.infer<typeof merged>} MergedOutput */
+/** @typedef {z.SafeParseReturnType<MergedInput, MergedOutput>} MergedSafeParseReturn */
+
+let env = /** @type {MergedOutput} */ (process.env);
+
+if (!!process.env.SKIP_ENV_VALIDATION == false) {
+  const isServer = typeof window === 'undefined';
+
+  const parsed = /** @type {MergedSafeParseReturn} */ isServer
+    ? merged.safeParse(processEnv) // on server we can validate all env vars
+    : client.safeParse(processEnv); // on client we can only validate the ones that are exposed
```

Comment:
This is going to fail for all the `z.boolean` and `z.number` values because all `process.env values are strings.